language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/backend/java/inner/RecordClass.scala

```diff
@@ -26,6 +27,14 @@ private[inner] object RecordClass extends StrictLogging {
       val recordType = TypeSpec
         .classBuilder(className)
         .addModifiers(Modifier.PUBLIC)
+        .superclass(
+          if (typeParameters.isEmpty)
+            ParameterizedTypeName.get(
+              ClassName get classOf[javaapi.data.codegen.DamlRecord[_]],
+              className,
+            )
+          else ClassName get classOf[Object]
```

Comment:
The goal is not to get every codegenned type to have a superclass, because e.g. primitives never will, but to make certain kinds of downstream abstractions in templates, interfaces, and choices simpler.

This restriction may mean we want _encoder_ types to complement decoder types eventually, but whether that becomes valuable may depend on future development of the language. For example, this is sufficient to help us implement choice encoders in metadata, but a broadening of the choice argument set might require more from codegen.